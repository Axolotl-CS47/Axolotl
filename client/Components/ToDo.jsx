import React, { useState, useEffect } from "react";
import { Country, City } from "country-state-city";
import store from "../store";
import { useSelector } from "react-redux";
const ToDo = () => {
  const [country, setCountry] = useState("");
  const [result, setResult] = useState("");
  const [countryCode, setCountryCode] = useState("");
  const [cities, setCities] = useState([]);
  const [citySelected, setCitySelected] = useState('')
  const [toDo, setToDo] = useState('');
  const [newArray, setNewArray] = useState([]);
  const allCountriesObj = Country.getAllCountries();
  const allCountries = allCountriesObj?.map(obj =>
    <option>{obj.name}</option>
  )


  const allTheCities = cities?.map(element => {
    return <option>{element}</option>
  })

  const arrToDo = []
  useEffect(() => {

    fetch(`https://api.opentripmap.com/0.1/en/places/geoname?name=${citySelected}&apikey=5ae2e3f221c38a28845f05b63e95fc09f0bc72578a5aa05f060e818b`)
    .then((res) => res.json())
    .then(data => {
        const lon = data.lon;
        const lat = data.lat; 
        return fetch(`https://api.opentripmap.com/0.1/en/places/radius?radius=1500&lon=${lon}&lat=${lat}&apikey=5ae2e3f221c38a28845f05b63e95fc09f0bc72578a5aa05f060e818b`)
    })
    .then(res => res.json())
    .then(data => {
      const arr = data.features?.map(element => element.properties.xid);
      const arrayTest = []
      for (let i = 0; i<5; i++) {
        fetch(`https://api.opentripmap.com/0.1/en/places/xid/${arr[i]}?apikey=5ae2e3f221c38a28845f05b63e95fc09f0bc72578a5aa05f060e818b`)
        .then(res => res.json())
        .then(dataXXID => {
         arrayTest.push(dataXXID)
         console.log('dataxxid', arrayTest)
        })
      }
    })
    .catch(err => console.log(err));      
  }, [citySelected])

  const allTheCities = cities.map((element) => {
    return <option>{element}</option>;
  });

  const handleClick = (value) => {
const todoSelect = document.querySelector('todo');
    const arr = allCountriesObj.filter((element) => {
      return element.name === value;
    });

    const code = arr[0].isoCode;
    const allCitiesObj = City.getAllCities();
    const allCities = allCitiesObj
      .filter((obj) => {
        return obj.countryCode === code;
      })
      .map((element) => {
        return element.name;
      });

    setCities([...allCities]);
  };
  return (
    <>
      <h1>Things to do in {`${country}`}</h1>
      <select
      id="todo"
        onChange={(e) => handleClick(e.target.value)}
      >
        {allCountries}
      </select>
      <select onChange={(e) => setCitySelected(e.target.value)}>
        <option>City</option>
        {allTheCities}
      </select>
      <input type = 'text' value={countrySelectedInMap.currentCountry}/>
    </>
  );
};

export default ToDo;
