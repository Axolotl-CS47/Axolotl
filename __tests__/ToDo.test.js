/**
 * @jest-environment jsdom
 */

 import React from 'react';
 import '@testing-library/jest-dom'
 import renderer from 'react-test-renderer';
import ToDo from '../client/components/ToDo';
 
 it('renders ToDo', async () => {
  const component = renderer.create(
    <ToDo />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
 });
 
