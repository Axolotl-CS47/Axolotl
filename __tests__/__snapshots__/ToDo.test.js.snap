// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ToDo 1`] = `
Array [
  <h1>
    Things to do in 
    
  </h1>,
  <hr />,
  <select
    id="todo"
    onChange={[Function]}
  >
    <option>
      Country
    </option>
    <option>
      Afghanistan
    </option>
    <option>
      Aland Islands
    </option>
    <option>
      Albania
    </option>
    <option>
      Algeria
    </option>
    <option>
      American Samoa
    </option>
    <option>
      Andorra
    </option>
    <option>
      Angola
    </option>
    <option>
      Anguilla
    </option>
    <option>
      Antarctica
    </option>
    <option>
      Antigua And Barbuda
    </option>
    <option>
      Argentina
    </option>
    <option>
      Armenia
    </option>
    <option>
      Aruba
    </option>
    <option>
      Australia
    </option>
    <option>
      Austria
    </option>
    <option>
      Azerbaijan
    </option>
    <option>
      Bahamas The
    </option>
    <option>
      Bahrain
    </option>
    <option>
      Bangladesh
    </option>
    <option>
      Barbados
    </option>
    <option>
      Belarus
    </option>
    <option>
      Belgium
    </option>
    <option>
      Belize
    </option>
    <option>
      Benin
    </option>
    <option>
      Bermuda
    </option>
    <option>
      Bhutan
    </option>
    <option>
      Bolivia
    </option>
    <option>
      Bosnia and Herzegovina
    </option>
    <option>
      Botswana
    </option>
    <option>
      Bouvet Island
    </option>
    <option>
      Brazil
    </option>
    <option>
      British Indian Ocean Territory
    </option>
    <option>
      Brunei
    </option>
    <option>
      Bulgaria
    </option>
    <option>
      Burkina Faso
    </option>
    <option>
      Burundi
    </option>
    <option>
      Cambodia
    </option>
    <option>
      Cameroon
    </option>
    <option>
      Canada
    </option>
    <option>
      Cape Verde
    </option>
    <option>
      Cayman Islands
    </option>
    <option>
      Central African Republic
    </option>
    <option>
      Chad
    </option>
    <option>
      Chile
    </option>
    <option>
      China
    </option>
    <option>
      Christmas Island
    </option>
    <option>
      Cocos (Keeling) Islands
    </option>
    <option>
      Colombia
    </option>
    <option>
      Comoros
    </option>
    <option>
      Congo
    </option>
    <option>
      Congo The Democratic Republic Of The
    </option>
    <option>
      Cook Islands
    </option>
    <option>
      Costa Rica
    </option>
    <option>
      Cote D'Ivoire (Ivory Coast)
    </option>
    <option>
      Croatia (Hrvatska)
    </option>
    <option>
      Cuba
    </option>
    <option>
      Cyprus
    </option>
    <option>
      Czech Republic
    </option>
    <option>
      Denmark
    </option>
    <option>
      Djibouti
    </option>
    <option>
      Dominica
    </option>
    <option>
      Dominican Republic
    </option>
    <option>
      East Timor
    </option>
    <option>
      Ecuador
    </option>
    <option>
      Egypt
    </option>
    <option>
      El Salvador
    </option>
    <option>
      Equatorial Guinea
    </option>
    <option>
      Eritrea
    </option>
    <option>
      Estonia
    </option>
    <option>
      Ethiopia
    </option>
    <option>
      Falkland Islands
    </option>
    <option>
      Faroe Islands
    </option>
    <option>
      Fiji Islands
    </option>
    <option>
      Finland
    </option>
    <option>
      France
    </option>
    <option>
      French Guiana
    </option>
    <option>
      French Polynesia
    </option>
    <option>
      French Southern Territories
    </option>
    <option>
      Gabon
    </option>
    <option>
      Gambia The
    </option>
    <option>
      Georgia
    </option>
    <option>
      Germany
    </option>
    <option>
      Ghana
    </option>
    <option>
      Gibraltar
    </option>
    <option>
      Greece
    </option>
    <option>
      Greenland
    </option>
    <option>
      Grenada
    </option>
    <option>
      Guadeloupe
    </option>
    <option>
      Guam
    </option>
    <option>
      Guatemala
    </option>
    <option>
      Guernsey and Alderney
    </option>
    <option>
      Guinea
    </option>
    <option>
      Guinea-Bissau
    </option>
    <option>
      Guyana
    </option>
    <option>
      Haiti
    </option>
    <option>
      Heard Island and McDonald Islands
    </option>
    <option>
      Honduras
    </option>
    <option>
      Hong Kong S.A.R.
    </option>
    <option>
      Hungary
    </option>
    <option>
      Iceland
    </option>
    <option>
      India
    </option>
    <option>
      Indonesia
    </option>
    <option>
      Iran
    </option>
    <option>
      Iraq
    </option>
    <option>
      Ireland
    </option>
    <option>
      Israel
    </option>
    <option>
      Italy
    </option>
    <option>
      Jamaica
    </option>
    <option>
      Japan
    </option>
    <option>
      Jersey
    </option>
    <option>
      Jordan
    </option>
    <option>
      Kazakhstan
    </option>
    <option>
      Kenya
    </option>
    <option>
      Kiribati
    </option>
    <option>
      Korea North
    </option>
    <option>
      Korea South
    </option>
    <option>
      Kuwait
    </option>
    <option>
      Kyrgyzstan
    </option>
    <option>
      Laos
    </option>
    <option>
      Latvia
    </option>
    <option>
      Lebanon
    </option>
    <option>
      Lesotho
    </option>
    <option>
      Liberia
    </option>
    <option>
      Libya
    </option>
    <option>
      Liechtenstein
    </option>
    <option>
      Lithuania
    </option>
    <option>
      Luxembourg
    </option>
    <option>
      Macau S.A.R.
    </option>
    <option>
      Macedonia
    </option>
    <option>
      Madagascar
    </option>
    <option>
      Malawi
    </option>
    <option>
      Malaysia
    </option>
    <option>
      Maldives
    </option>
    <option>
      Mali
    </option>
    <option>
      Malta
    </option>
    <option>
      Man (Isle of)
    </option>
    <option>
      Marshall Islands
    </option>
    <option>
      Martinique
    </option>
    <option>
      Mauritania
    </option>
    <option>
      Mauritius
    </option>
    <option>
      Mayotte
    </option>
    <option>
      Mexico
    </option>
    <option>
      Micronesia
    </option>
    <option>
      Moldova
    </option>
    <option>
      Monaco
    </option>
    <option>
      Mongolia
    </option>
    <option>
      Montenegro
    </option>
    <option>
      Montserrat
    </option>
    <option>
      Morocco
    </option>
    <option>
      Mozambique
    </option>
    <option>
      Myanmar
    </option>
    <option>
      Namibia
    </option>
    <option>
      Nauru
    </option>
    <option>
      Nepal
    </option>
    <option>
      Bonaire, Sint Eustatius and Saba
    </option>
    <option>
      Netherlands The
    </option>
    <option>
      New Caledonia
    </option>
    <option>
      New Zealand
    </option>
    <option>
      Nicaragua
    </option>
    <option>
      Niger
    </option>
    <option>
      Nigeria
    </option>
    <option>
      Niue
    </option>
    <option>
      Norfolk Island
    </option>
    <option>
      Northern Mariana Islands
    </option>
    <option>
      Norway
    </option>
    <option>
      Oman
    </option>
    <option>
      Pakistan
    </option>
    <option>
      Palau
    </option>
    <option>
      Palestinian Territory Occupied
    </option>
    <option>
      Panama
    </option>
    <option>
      Papua new Guinea
    </option>
    <option>
      Paraguay
    </option>
    <option>
      Peru
    </option>
    <option>
      Philippines
    </option>
    <option>
      Pitcairn Island
    </option>
    <option>
      Poland
    </option>
    <option>
      Portugal
    </option>
    <option>
      Puerto Rico
    </option>
    <option>
      Qatar
    </option>
    <option>
      Reunion
    </option>
    <option>
      Romania
    </option>
    <option>
      Russia
    </option>
    <option>
      Rwanda
    </option>
    <option>
      Saint Helena
    </option>
    <option>
      Saint Kitts And Nevis
    </option>
    <option>
      Saint Lucia
    </option>
    <option>
      Saint Pierre and Miquelon
    </option>
    <option>
      Saint Vincent And The Grenadines
    </option>
    <option>
      Saint-Barthelemy
    </option>
    <option>
      Saint-Martin (French part)
    </option>
    <option>
      Samoa
    </option>
    <option>
      San Marino
    </option>
    <option>
      Sao Tome and Principe
    </option>
    <option>
      Saudi Arabia
    </option>
    <option>
      Senegal
    </option>
    <option>
      Serbia
    </option>
    <option>
      Seychelles
    </option>
    <option>
      Sierra Leone
    </option>
    <option>
      Singapore
    </option>
    <option>
      Slovakia
    </option>
    <option>
      Slovenia
    </option>
    <option>
      Solomon Islands
    </option>
    <option>
      Somalia
    </option>
    <option>
      South Africa
    </option>
    <option>
      South Georgia
    </option>
    <option>
      South Sudan
    </option>
    <option>
      Spain
    </option>
    <option>
      Sri Lanka
    </option>
    <option>
      Sudan
    </option>
    <option>
      Suriname
    </option>
    <option>
      Svalbard And Jan Mayen Islands
    </option>
    <option>
      Swaziland
    </option>
    <option>
      Sweden
    </option>
    <option>
      Switzerland
    </option>
    <option>
      Syria
    </option>
    <option>
      Taiwan
    </option>
    <option>
      Tajikistan
    </option>
    <option>
      Tanzania
    </option>
    <option>
      Thailand
    </option>
    <option>
      Togo
    </option>
    <option>
      Tokelau
    </option>
    <option>
      Tonga
    </option>
    <option>
      Trinidad And Tobago
    </option>
    <option>
      Tunisia
    </option>
    <option>
      Turkey
    </option>
    <option>
      Turkmenistan
    </option>
    <option>
      Turks And Caicos Islands
    </option>
    <option>
      Tuvalu
    </option>
    <option>
      Uganda
    </option>
    <option>
      Ukraine
    </option>
    <option>
      United Arab Emirates
    </option>
    <option>
      United Kingdom
    </option>
    <option>
      United States
    </option>
    <option>
      United States Minor Outlying Islands
    </option>
    <option>
      Uruguay
    </option>
    <option>
      Uzbekistan
    </option>
    <option>
      Vanuatu
    </option>
    <option>
      Vatican City State (Holy See)
    </option>
    <option>
      Venezuela
    </option>
    <option>
      Vietnam
    </option>
    <option>
      Virgin Islands (British)
    </option>
    <option>
      Virgin Islands (US)
    </option>
    <option>
      Wallis And Futuna Islands
    </option>
    <option>
      Western Sahara
    </option>
    <option>
      Yemen
    </option>
    <option>
      Zambia
    </option>
    <option>
      Zimbabwe
    </option>
    <option>
      Kosovo
    </option>
    <option>
      Curaçao
    </option>
    <option>
      Sint Maarten (Dutch part)
    </option>
  </select>,
  <select
    id="todo"
    name="select-city"
    onChange={[Function]}
  >
    <option>
      City
    </option>
  </select>,
  <table
    className="travelInfo"
  >
    
  </table>,
]
`;
